tests/src/test/scala/cats/tests/FoldableTests.scala

```diff
@@ -235,3 +241,44 @@ class FoldableValidatedCheck extends FoldableCheck[Validated[String, ?]]("valida
 class FoldableTryCheck extends FoldableCheck[Try]("try") {
   def iterator[T](tryt: Try[T]): Iterator[T] = tryt.toOption.iterator
 }
+
+class FoldableEitherKCheck extends FoldableCheck[EitherK[Option, Option, ?]]("eitherK") {
+  def iterator[T](eitherK: EitherK[Option, Option, T]) = eitherK.run.bimap(_.iterator, _.iterator).merge
+}
+
+class FoldableIorCheck extends FoldableCheck[Int Ior ?]("ior") {
+  def iterator[T](ior: Int Ior T) =
+    ior.fold(_ => None.iterator, b => Some(b).iterator, (_, b) => Some(b).iterator)
+}
+
+class FoldableIdCheck extends FoldableCheck[Id[?]]("id") {
+  def iterator[T](id: Id[T]) = Some(id).iterator
+}
+
+class FoldableIdTCheck extends FoldableCheck[IdT[Option, ?]]("idT") {
+  def iterator[T](idT: IdT[Option, T]) = idT.value.iterator
+}
+
+class FoldableConstCheck extends FoldableCheck[Const[Int, ?]]("const") {
+  def iterator[T](const: Const[Int, T]) = None.iterator
+}
+
+class FoldableTuple2Check extends FoldableCheck[(Int, ?)]("tuple2") {
+  def iterator[T](tuple: (Int, T)) = Some(tuple._2).iterator
+}
+
+class FoldableOneAndCheck extends FoldableCheck[OneAnd[List, ?]]("oneAnd") {
```

Comment:
I agree there is some duplication, but I don't think it should be tackled in this issue and should probably a different issue where we discuss what to do with tests for a typeclass which aren't part of the discipline laws.